File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    public Object solution(int n, int[][] computers) {
        boolean[] visit = new boolean[n];
        int answer = 0;
        
        for ( int i = 0 ; i < n ; i++ ) {
            if ( !visit[i] ) {
                dfs(i, computers, visit);
                answer++;
            }
        }
        
        return answer;
    }
    
    private void dfs(int r, int[][] computers, boolean[] visit) {
        visit[r] = true;
        for ( int i = 0 ; i < computers.length ; i++ ) {
            if ( computers[r][i] == 1 && !visit[i] ) {
                dfs(i, computers, visit);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    p = list(range(n))
    def f(x):
        if p[x] != x: p[x] = f(p[x])
        return p[x]
    for i in range(n):
        for j in range(n):
            if computers[i][j]: p[f(j)] = f(i)
    return len({f(i) for i in range(n)})